src/SimpleAmqpClient/Channel.h

```diff
@@ -89,9 +90,11 @@ class SIMPLEAMQPCLIENT_EXPORT Channel : boost::noncopyable
                         const std::string &username = "guest",
                         const std::string &password = "guest",
                         const std::string &vhost = "/",
-                        int frame_max = 131072)
+                        int frame_max = 131072,
+                        const amqp_sasl_method_enum sasl_method=AMQP_SASL_METHOD_PLAIN)
```

Comment:
- Don't change this function signature.
- Add new constructors that allow the user to authenticate using the EXTERNAL mechanism.
- The SimpleAmqpClient headers should be useable without needing the rabbitmq-c headers installed (IOW: don't `#include <amqp.h>` in the public SimpleAmqpClient headers or use things defined in the rabbitmq-c headers in the public SimpleAmqpClient headers.  It is ok to use these in the .cpp and non-public SimpleAmqpClient headers).